lightning_client.go

```diff
@@ -3085,6 +3086,33 @@ func (p *ChannelClosedUpdate) CloseTxid() chainhash.Hash {
 	return p.CloseTx
 }
 
+// CloseChannelOption is a functional type for an option that modifies a
+// CloseChannelRequest.
+type CloseChannelOption func(r *lnrpc.CloseChannelRequest)
+
+// SatPerVbyte is an option for setting the fee rate of a CloseChannelRequest.
+func SatPerVbyte(satPerVbyte uint64) CloseChannelOption {
+	return func(r *lnrpc.CloseChannelRequest) {
+		r.SatPerVbyte = satPerVbyte
+	}
+}
+
+// MaxFeePerVbyte is an option for setting the maximum fee rate a closer is
+// willing to pay on a CloseChannelRequest.
+func MaxFeePerVbyte(maxFeePerVbyte uint64) CloseChannelOption {
+	return func(r *lnrpc.CloseChannelRequest) {
+		r.MaxFeePerVbyte = maxFeePerVbyte
+	}
+}
+
+// WithNoWait is an option for setting the NoWait flag on an
+// CloseChannelRequest.
+func WithNoWait() CloseChannelOption {
```

Comment:
If only Go would have namespaces so we don't pollute the global namespace with these generic names. (no action required, just a rant)